include/nbl/builtin/hlsl/colorspace.hlsl

```diff
@@ -0,0 +1,198 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_COLOR_SPACE_COLORSPACE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_COLOR_SPACE_COLORSPACE_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace colorspace
+{
+
+struct colorspace_base
+{
+    NBL_CONSTEXPR_STATIC_INLINE float32_t wavelength_R = 580.0f;
+    NBL_CONSTEXPR_STATIC_INLINE float32_t wavelength_G = 550.0f;
+    NBL_CONSTEXPR_STATIC_INLINE float32_t wavelength_B = 450.0f;
+};
+
+struct scRGB : colorspace_base
+{
+    static float32_t3x3 FromXYZ()
+    {
+        return float32_t3x3(
+            float32_t3( 3.240970f, -1.537383f, -0.498611f),
+            float32_t3(-0.969244f,  1.875968f,  0.041555f),
+            float32_t3( 0.055630f, -0.203977f,  1.056972f)
+        );
+    }
+    static float32_t3 FromXYZ(float32_t3 val) { return hlsl::mul(FromXYZ(), val); }
+
+    static float32_t3x3 ToXYZ()
+    {
+        return float32_t3x3(
+            float32_t3(0.412391f, 0.357584f, 0.180481f),
+            float32_t3(0.212639f, 0.715169f, 0.072192f),
+            float32_t3(0.019331f, 0.119195f, 0.950532f)
+        );
```

Comment:
why are you copying literals ? return the encode/decode matrices we already have!